packages/test/test-end-to-end-tests/src/test/batchBreakRegression.spec.ts

```diff
@@ -0,0 +1,351 @@
+        }
+        assert.equal(testObject.root.get("foo"), "bar");
+    }
+}
+
+describeNoCompat("Batching failures", (getTestObjectProvider) => {
+    it("working proxy",
+    async function() {
+        const provider = getTestObjectProvider(true);
+
+        const proxyDsf = createFunctionOverrideProxy<IDocumentServiceFactory>(
+            provider.documentServiceFactory,
+            {
+                createDocumentService: {
+                    connectToDeltaStream: {
+                        submit: (ds) => (messages) => {
+                            // validate a no-op proxy works
+                            ds.submit(messages);
+                        },
+                    },
+                },
+            });
+        await runAndValidateBatch(provider, proxyDsf, this.timeout());
+    });
+    it("working batch",
+    async function() {
+        const provider = getTestObjectProvider(true);
+        await runAndValidateBatch(provider, provider.documentServiceFactory, this.timeout());
+    });
+    describe("client sends invalid batches ", () => {
+        itExpects("Batch end without start",
+        [
+            { eventName: "fluid:telemetry:Container:ContainerClose", error: "OpBatchIncomplete" },
+        ],
+        async function() {
+            const provider = getTestObjectProvider(true);
+
+            const proxyDsf = createFunctionOverrideProxy<IDocumentServiceFactory>(
+                provider.documentServiceFactory,
+                {
+                    createDocumentService: {
+                        connectToDeltaStream: {
+                            submit: (ds) => (messages) => {
+                                const newMessages = [...messages];
+                                const batchStartIndex = newMessages.findIndex((m) => m.metadata?.batch === true);
+                                if (batchStartIndex >= 0) {
+                                    newMessages[batchStartIndex] = {
+                                        ... newMessages[batchStartIndex],
+                                        metadata: {
+                                            ... newMessages[batchStartIndex].metadata,
+                                            batch: undefined,
+                                        },
+                                    };
+                                }
+                                ds.submit(newMessages);
+                            },
+                        },
+                    },
+                });
+
+            try {
+                await runAndValidateBatch(provider, proxyDsf, this.timeout());
+                assert.fail("expected error");
+            } catch (e) {
+                assert(isILoggingError(e), `${e}`);
+                assert.equal(e.message, "OpBatchIncomplete", e);
+            }
+        });
+
+        // bug bug: container runtime never unpauses if there is no batch end
+        itExpects.skip("Batch start without end",
```

Comment:
@andre4i  not sure we care about this case. we'd never know if we hit it as the container just hangs forever

Reply:
we hang forever as we never unpause the incoming queue

Reply:
i'm curious why we even need to pause here. my guess is there are cases where, like maybe loading from storage where we don't want to prevent inteleaving, if so, this behavior still seems wrong, as it seems we allow js turns between batches in some cases then. instead we probably want to hold all messages until have the full batch before dispatching synchronously

Reply:
The queue is paused so that ops are not processed and sent to rest of the system until we receive a full batch. Another reason to pause is so that we can yield op processing if we process for too long.
I agree with you that pausing is not the best solution here. We can store incomplete batches in delta queue until we receive it in full and then release to the rest of the system. That will make is much simpler and less error prone.